client/state/data-layer/wpcom/sites/comments/index.js

```diff
@@ -138,31 +139,43 @@ export const receiveCommentError = ( { dispatch, getState }, { siteId, commentId
 
 // @see https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/comments/
 export const fetchCommentsList = ( { dispatch }, action ) => {
-	const { query } = action;
-
-	if ( ! query || 'site' !== query.listType ) {
+	if ( ! action.query || 'site' !== action.query.listType ) {
 		return;
 	}
 
-	const { siteId, status = 'unapproved', type = 'comment' } = query;
+	const { siteId, status = 'unapproved', type = 'comment' } = action.query;
+
+	const query = {
+		...omit( action.query, [ 'listType', 'siteId' ] ),
+		status,
+		type,
+	};
 
 	dispatch(
 		http(
 			{
 				method: 'GET',
 				path: `/sites/${ siteId }/comments`,
 				apiVersion: '1.1',
-				query: {
-					status,
-					type,
-				},
+				query,
 			},
 			action,
 		),
 	);
 };
 
-export const addComments = ( { dispatch }, { query: { siteId } }, next, { comments } ) => {
+export const addComments = ( { dispatch }, { query: { siteId, status } }, next, { comments } ) => {
+	// Initialize the comments tree
```

Comment:
For the whole `isLoading` / `emptyContent` handling of Comment Management lists, we set `comments.treesInitialized` (so that we know that an empty tree is actually empty and not loading) by listening to `COMMENTS_TREE_SITE_ADD`, which normally is dispatched on `fetchTree` success.

In case of JP sites, we never call `fetchTree`, but we populate the trees as a side effect of `fetchCommentsList`, listening to `COMMENTS_RECEIVE`.

If we receive even a single comment, we don't need to set `treesInitialized`, because a non-empty tree is good enough for the various tree selectors.

But if we receive an empty response, we need to "manually" initialize the tree, so that the tree selectors actually know that is loaded and empty.